### ISSUE TITLE
Confusion about dataset split

### ISSUE THREAD

**Issue by wasim004** [nii-yamagishilab/Capsule-Forensics-v2#13]
Hi,
Can you please explain Table.1?
I mean what does `720*3` vids, `140*3` vid and `140*3` represents for all train, val and test datasets?
I am unable to understand the video split?
Secondly, can you please explain how you dealt with `images` data and `videos` data? How you placed them in folders to prepare dataloader? 
Also, let me confirm about whether you extract frames from videos and make a videos again from those extracted frames of videos and used `test_binary_vid_ffpp.py` OR `test_multiclass_vid_ffpp.py` to evaluate the model on test videos?
Thanks!

**Comment by honghuy127** [comment]
Hi,

The train/val/test set split annotations are from [here](https://github.com/ondyari/FaceForensics/tree/master/dataset/splits). The number 3 means we used 3 compression levels: c0, c23, and c40 for each video.

All of our experiments used video frames extracted by these [scripts](https://github.com/nii-yamagishilab/Capsule-Forensics-v2/tree/master/datascripts).
About `test_binary_vid_ffpp.py` and `test_multiclass_vid_ffpp.py`, these scripts used image frames, not video files. Frames from the same video have similar names (the difference is in the counting part).

**Comment by wasim004** [comment]
Hi,

Thanks for your kind response.
I got it now.

Thanks!

**Comment by wasim004** [comment]
Hi,
For images classification, did you followed the directory structure like e-g `train/0_original/*.jpeg`?
Because my directory structure is like `train-test-validation/0_original-1_deepfake-etc/video-folder/concerened-frames-of-this-video`
e-g `/databases/faceforensicspp/train/0_original/000/frame_det_00_000001.jpeg/`
![image](https://user-images.githubusercontent.com/24964488/158539581-781acc50-46b5-42c3-91fc-90786a03e5b6.png)



Thanks!

**Comment by honghuy127** [comment]
For dataloader, we used `torchvision.datasets.ImageFolder`, so you need to put the images in `0_original` without subfolders.

You can also see the discussion [here](https://github.com/nii-yamagishilab/Capsule-Forensics-v2/issues/8).

**Comment by wasim004** [comment]
Hi,
Thanks you for response!
Well, so for videos you also directly placed the images inside the `0_original` or `1_deepfake` etc?
If yes, how the model knows which video the frames belongs to?
Thanks!

**Comment by honghuy127** [comment]
Yes, we treat a video as a sequence of numbered images.
The scripts `test_vid_binary_ffpp.py` and `test_vid_multiclass_ffpp.py` decide whether images belong to a videos based on their name.

**Comment by wasim004** [comment]
Fine, Thanks You!
BTW can you please explain the `Real` column in below table? I mean what these scores shows?
Thanks!

![image](https://user-images.githubusercontent.com/24964488/158574971-33f9a57a-4d4f-4c17-a919-e37ea964dc01.png)

**Comment by honghuy127** [comment]
Real means original images.
The numbers there are accuracies. For instance, in 'real' column, they are ratios of correct 'real' predictions to all 'real' samples.

**Comment by wasim004** [comment]
Fine, Thanks!

**Comment by wasim004** [comment]
Hi,

May I get the code for visualizing the capsule layers using GradCam or Guided Backpropagation in your case. I need to select the layer features of the capsule network.
Thanks!

**Comment by honghuy127** [comment]
Hi,

I used the code from [this repository](https://github.com/utkuozbulak/pytorch-cnn-visualizations) for visualization.
You need to modify the code of the model to make it work.
